background_worker/jobs/clear_block_transactions_map_test.go

```diff
@@ -24,36 +25,30 @@ func (s *ClearBlockTransactionsMapSuite) Test() {
 		RecordRetentionDays: 10,
 	}
 
-	// Add "fresh" blocks
-	for i := 0; i < 5; i++ {
-		m := store.BlockTransactionMap{
-			BlockID:       int64(i),
-			TransactionID: int64(i),
-			Pos:           int64(i),
-		}
-		m.InsertedAt = time.Now()
-		s.InsertBlockTransactionMap(&m)
-	}
+	db, err := sqlx.Open("postgres", params.String())
+	require.NoError(s.T(), err)
 
-	// Add "expired" blocks
-	for i := 0; i < 5; i++ {
-		m := store.BlockTransactionMap{
-			BlockID:       10 + int64(i),
-			TransactionID: 10 + int64(i),
-			Pos:           10 + int64(i),
-		}
-		m.InsertedAt = time.Now().Add(-20 * 24 * time.Hour)
-		s.InsertBlockTransactionMap(&m)
-	}
+	fixtures, err := testfixtures.New(
+		testfixtures.Database(db.DB),
+		testfixtures.Dialect("postgresql"),
+		testfixtures.Directory("fixtures"), // The directory containing the YAML files
+	)
+	require.NoError(s.T(), err)
 
-	db, err := sqlx.Open("postgres", params.String())
+	err = fixtures.Load()
 	require.NoError(s.T(), err)
 
-	err = ClearBlockTransactionsMap(params)
+	now := time.Date(2023, 12, 22, 12, 0, 0, 0, time.UTC)
```

Comment:
you don't need to manipulate date inside ClearJob. 
You simply insert "old" records into db and make sure it's deleted after job is complete.